src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -102,7 +103,7 @@ public class BuildManager : INodePacketHandler, IBuildComponentHost, IDisposable
         /// <summary>
         /// Set of active nodes in the system.
         /// </summary>
-        private readonly HashSet<NGen<int>> _activeNodes;
+        private readonly ConcurrentDictionary<int, byte> _activeNodes;
```

Comment:
To minimize the change, and to not give false impressions that BuildManager handles packets in parallel, you can avoid making this a concurrent data structure by hiding the parallel node creation with local concurrent data structures which you then dump into `_activeNodes` after all nodes get created.